### PR TITLE
Soot setup

### PR DESCRIPTION
**Pull request by ftc**
https://gitlab.com/cuplv/csci7135-s17/issues/34

### CODE REVIEW

**Comment by ftc** [comment]
The problem with soot here is that the Jimple body is only available in certain packs and phases.  Usually a program analysis goes into the wjtp package and results are tagged onto the source code for usage later.  I wrote a thin wrapper that lets us call something and get a result but there are additional problems.  Since soot relies on a global static Scene instance (Scene.v) it can only be run once which will cause problems for our unit tests.  There is probably a better way to do this but I need to move on for today.  I will talk about these issues on monday.

File: src/main/scala/Test.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+  * Created by s on 3/12/17.
```
Comment (bechang) [inline src/main/scala/Test.scala:2]: Could you change your IntelliJ template to include your in scaladoc style? For example, I have

```
/**
  * @author Bor-Yuh Evan Chang
  */
```

**Review by bechang — CHANGES_REQUESTED** [review]
Could we figure out a better to connect with Soot without commiting jars?

**Comment by tianhan0** [comment]
Most of the code is fine. Two minor problems.

I took Shawn's branch (soot-setup) and tested on my machine, it didn't work at first. It turns out that he hard coded the java library path with path separator "/". But since I'm using Windows, I made a little bit of modification to make code more compatible.

Meanwhile, "sys.env("JAVA_HOME")" didn't work on Windows somehow. So I changed it to "System.getProperty("java.home")", which seems to be more portable on all platforms.

**Comment by bechang** [comment]
@litho17 So how do we get rid of the JAVA_HOME all together?

**Comment by tianhan0** [comment]
@bechang According to [Oracle Java Documentation](https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html), **System.getProperty("java.home")** should be compatible on all platforms.

**Pull request closed without merging**